docs/cluster-management/configure-buildcluster-queue-worker.md

```diff
@@ -0,0 +1,753 @@
+      url: "#configure-build-cluster-queue-worker"
+    - title: Build Cluster Schema Definition
+      url: "#build-cluster-schema-definition"
+---
+It is Cluster admins responsibility to setup and configure Rabbitmq Message Broker and Build Cluster Queue Worker.
+
+# Managing the Build Cluster Queue Worker
+
+This page will cover how to setup [RabbitMQ Message Broker](https://www.rabbitmq.com/#getstarted) and [Build Cluster Queue Worker](https://github.com/screwdriver-cd/buildcluster-queue-worker). 
+
+## Overview
+
+Build cluster feature can be enabled/disabled by [multiBuildCluster feature flag](https://github.com/screwdriver-cd/screwdriver/blob/master/config/default.yaml#L257) or using 
+[environment variable](https://github.com/screwdriver-cd/screwdriver/blob/master/config/custom-environment-variables.yaml#L369).
+When enabled Screwdriver [Queue Service](./configure-queue-service.md) will push build message to Rabbitmq exchange. Build message header will be set with routing key based on an active flag and 
+weightage defined in buildClusters table. Rabbitmq exchange will route build message to queue based on routing key defined in message header and build message will be consumed and processed by 
+Build Cluster Queue Worker. For build cluster stickiness, when first build is run for a pipeline, build cluster routing key will be added to pipeline annotations. Build cluster stickiness will 
+not impact when a build cluster is taken offline for maintenance as builds will be automatically routed to next available build cluster. 
+
+***Caution**: Please give attention when creating buildCluster routing keys and be cautious or refrain updating routing key. Routing key is added to pipeline annotations for build cluster stickiness. 
+So any updates to routing key, need to be cascaded to affected pipelines annotations and rabbitmq exchange bindings. Without this update, builds related to affected pipeline will error. 
+Another option is to create a new build cluster and disable old build cluster. In this case builds will auto route to new build cluster without any updates, but please keep in mind build cluster 
+stickiness will be lost.* 
+
+## Setup Build Cluster
+
+Cluster admin should create build cluster using [buildclusters API](https://api.screwdriver.cd/v4/documentation#/v4/postV4Buildclusters)
+
+```json
+{
+  "name": "ClusterA",
+  "description": "Build cluster to process ClusterA builds",
+  "scmOrganizations": [
+    "screwdriver-cd"
+  ],
+  "isActive": true,
+  "managedByScrewdriver": true,
+  "maintainer": "foo@bar.com",
+  "weightage": 100
+}
+```
+
+1. name should match the queue and routing key defined in rabbitmq configuration.
+1. set `managedByScrewdriver` to `true`, if build cluster is managed by internal Screwdriver team.
+1. set `isActive` to `true` or `false` to turn on/off a build cluster.
+1. set weightage 100 if you have single build cluster and for more than one cluster, distribute weightage accordingly.
```

Comment:
we had an [issue](https://github.com/screwdriver-cd/screwdriver/issues/2307) for multiple scms in multiple build clusters.
The [scmContext](https://github.com/screwdriver-cd/data-schema/blob/master/models/buildCluster.js#L30) of buildClusters is decided by the API Token which is belonging to the specific scm.
And if we want to build in the build cluster with another scm, we have to post `/v4/buildClusters` with same data by the another token which is issued by another scm account. it’s a little confusing so i'm glad if it has a caution for that.

Reply:
doc updated